interface/docker_build_interface.sh

```diff
@@ -1,17 +1,37 @@
+script_directory=`dirname "${BASH_SOURCE[0]}"  | xargs realpath`
+
+cd $script_directory
+
 docker build -t adage-server/docker-interface-base -f Dockerfile.base .
 
 docker build -t adage-server/docker-interface-build -f Dockerfile .
 
-# Set up mounting volume directory that will house bin and build folders
-# if it doesn't already exist.
-script_directory=`dirname "${BASH_SOURCE[0]}"  | xargs realpath`
 compile_folder="$script_directory/compile_folder"
+bin_folder="$script_directory/bin"
+build_folder="$script_directory/build"
 
-if [ ! -d "$compile_folder" ]; then
+# Set up mounting volume directory that will house bin and build folders
+# if it doesn't already exist.
+if [ -d "$compile_folder" ]
```

Comment:
Do you think it makes more sense to use `-e` instead of `-d`? If `$compile_folder` exists but is not a directory, the `mkdir $compile_folder` command on line 19 will fail.

Reply:
Hmm, good question. I don't know if `-e` would be better than `-d` here since we really only expect the compile_folder, bin_folder and build_folder to be directories that were previously built by out interface build process. I think if they were something other than directories, that would be kind of unexpected, and maybe we would want this to fail. What scenarios were you thinking of where these would be files that were not directories?

Reply:
Is it possible that these directories could be soft links? If they must be directories, then it is okay to use `-d` here, because the error generated by `mkdir` command will remind the user of this requirement.

Reply:
We are setting this up ourselves, so we know for sure if `compile_folder` is a real directory or a symlink (see the `mkdir` at line 19 below). I'd say the `-d` is exactly what we want.

Reply:
Right, ok, that makes sense - I'll leave it with `-d` then.